custom/sensors/SparkEncoder.java

```diff
@@ -0,0 +1,134 @@
+package org.usfirst.frc4904.standard.custom.sensors;
+
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import org.usfirst.frc4904.standard.Util;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class SparkEncoder implements CustomEncoder {
+    protected final CANEncoder encoder;
+    protected PIDSourceType pidSource;
+    protected double distancePerPulse;
+    protected boolean reverseDirection;
+    protected double truePosition;
+    protected double trueVelocity;
+
+    public SparkEncoder(CANSparkMax spark, boolean reverseDirection, double distancePerPulse) {
+        encoder = new CANEncoder(spark);
+        this.reverseDirection = reverseDirection;
+        this.distancePerPulse = distancePerPulse;
+        setPIDSourceType(PIDSourceType.kDisplacement);
+    }
+
+    @Override
+    public void setPIDSourceType(PIDSourceType pidSource) {
+        this.pidSource = pidSource;
+    }
+
+    @Override
+    public PIDSourceType getPIDSourceType() {
+        return pidSource;
+    }
+
+    @Override
+    public double pidGet() {
+        if (pidSource == PIDSourceType.kDisplacement) {
+            return getDistance();
+        } else {
+            return getRate();
+        }
+    }
+
+    @Override
+    public int get() {
```

Comment:
encoder returns data as a double representing rotations, so it wouldn't really make sense for this method to exist. Have we ever used get? should we modify it to fit this type of sensor as well (in CustomEncoder) or delete it?

Reply:
yeah, it never should be used, but its in custom encoder